pkg/server/preprocessor/preprocessor.go

```diff
@@ -142,12 +142,12 @@ func (p InjectPublicDatabase) Preprocess(payload *router.Payload, response *rout
 	return http.StatusOK
 }
 
-type RequireUserForWrite struct {
+type RequireUser struct {
 }
 
-func (p RequireUserForWrite) Preprocess(payload *router.Payload, response *router.Response) int {
+func (p RequireUser) Preprocess(payload *router.Payload, response *router.Response) int {
 	if payload.UserInfo == nil {
-		response.Err = skyerr.NewError(skyerr.PermissionDenied, "write is not allowed")
+		response.Err = skyerr.NewError(skyerr.PermissionDenied, "User is required for this action, please login.")
```

Comment:
Interesting. The error code should be `NotAuthenticated` rather than `PermissionDenied`. See `skyerr` for details.